src/System.Security.Principal.Windows/tests/WindowsIdentityTests.Impersonate.cs

```diff
@@ -0,0 +1,709 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+// We skip all tests on Uap because we need to create some new Windows users before run
+// On Uap sandbox we cannot create users
+[SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+public class WindowsIdentityTestsImpersonate : IClassFixture<WindowsIdentityImpersonateFixture>
+{
+    private const string authenticationType = "WindowsAuthentication";
+    private WindowsIdentityImpersonateFixture _fixture;
+
+    public WindowsIdentityTestsImpersonate(WindowsIdentityImpersonateFixture windowsIdentityImpersonateFixture)
+    {
+        _fixture = windowsIdentityImpersonateFixture;
+
+        Assert.False(_fixture.TestAccount1.AccountTokenHandle.IsInvalid);
+        Assert.False(string.IsNullOrEmpty(_fixture.TestAccount1.AccountName));
+
+        Assert.False(_fixture.TestAccount2.AccountTokenHandle.IsInvalid);
+        Assert.False(string.IsNullOrEmpty(_fixture.TestAccount2.AccountName));
+    }
+
+    // Test to do in a domain without SeTcbPrivilege privilege
+    // [Fact]
```

Comment:
What's the purpose / plan for these commented out [Fact]s?  Right now they're dead code.

Reply:
Because this tests are possible only inside a domain, I manually tested inside my office domain, but I've taked with @danmosemsft and he said that maybe some of you would re-test inside other domain, CI cannot test this cases. I mean this is manual tests.

Reply:
Crypto's solution to this: https://github.com/dotnet/corefx/blob/77096755443730962810662501f82550fdb4656d/src/System.Security.Cryptography.X509Certificates/tests/TestEnvironmentConfiguration.cs#L11-L12

Along with marking the tests as `[ConditionalFact(nameof(RunManualTests))]

Reply:
Of course, these tests require a bit more than uncommenting in order to work.